### PR TITLE
Port of optimum-setup - environments to docs

### PR DESCRIPTION
**Pull request by harrisonmeister**
This work involves a port of the environments section of the [Optimum Setup Book](https://github.com/OctopusDeploy/OptimumSetupBook/blob/master/manuscript/Environments.md)
- Split out a section in the existing  `environments` include as it contained items I wanted to place something between.
- Added new include for optimum-setup recommendations for environments. 
- Added new environment recommendations page
- Added a typical environment configuration section on the environments page

### CODE REVIEW

File: docs/getting-started-guides/environments.md [hunk 1]
```diff
@@ -8,4 +8,12 @@ position: 50
 
 !include <environments>
 
+!include <add-new-environments>
+
+:::hint
+Always try to keep the number of environments low, under a dozen or so where possible. Doing so helps with configuring other areas of Octopus, such as Lifecycles, Channels and Security. They also keep your dashboard easy to follow.
```
Comment (ryanrousseau) [inline docs/getting-started-guides/environments.md:14]: Re: Lee's comments, probably can trim these sentences down

> In general, keep the number of environments under ten. Having fewer environments makes configuring and maintaining your Octopus Server easier.

**Review by octocrock — DISMISSED** [review]
Tiny change...

File: docs/infrastructure/environments/index.md [hunk 2]
```diff
@@ -6,9 +6,26 @@ position: 10
 
 !include <environments>
 
+## Environment configuration
+
+Since environments are the phases that you move your code through, it forms the backbone of your deployment pipeline. Before you configure anything else, you should configure your environments.
```
[the remark this answers is not in the document]
Comment (harrisonmeister) [reply docs/infrastructure/environments/index.md]: Fixed in 5a844527264021acf7e197dfc00e2e6864571158 thanks! :+1:

File: docs/getting-started-guides/environments.md (hunk 1 above)
Comment (harrisonmeister) [reply docs/getting-started-guides/environments.md]: Fixed in 9954480c9f0e2fcfcff8767f5a63f7bc8af0c970, thanks!

File: docs/infrastructure/deployment-targets/dynamic-infrastructure/index.md [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
+description: Octopus resources created in cloud providers can also be modeled in Octopus using service messages and scripts, which form part of dynamic infrastructure.
 position: 120
 ---
 
-Some resources can be created within Octopus from the same scripts that you use to create them on Azure. By adding some additional commands Web Apps you create on Azure can also be created within Octopus as deployment targets.
+You can use the [Octopus REST API](/docs/octopus-rest-api/index.md) or the Octopus commands below to create Octopus accounts, targets, and workers dynamically. You can make these requests in the same scripts that create your cloud infrastructure or in following steps.
 
 :::success
-Only Azure Service Principal Accounts, Azure Web Apps, Azure Service Fabric, Azure Cloud Services and Kubernetes targets are supported.
+Only a subset of account types and deployment targets support being created dynamically. The ones which are supported are listed below.
```
Comment (ryanrousseau) [inline docs/infrastructure/deployment-targets/dynamic-infrastructure/index.md:10]: This is only true if you're using the commandlets, with the API you can create anything right?

Comment (harrisonmeister) [reply docs/infrastructure/deployment-targets/dynamic-infrastructure/index.md:10]: Correct, I'll add some clarification to that statement now I've added the API section

Comment (harrisonmeister) [reply docs/infrastructure/deployment-targets/dynamic-infrastructure/index.md]: Updated the success to a warning, and moved closer to the section it refers to (and added reference to the commands in) 1349e6c31247d60a26669b0e4389a541589db5aa

File: docs/shared-content/infrastructure/add-new-environments.include.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+## Add new environments {#add-new-environments}
+
+1. Navigate to **{{Infrastructure,Environments}}** and click **ADD ENVIRONMENT**.
+1. Give your new environment a meaningful name and click **SAVE**.
```
[the remark this answers is not in the document]
Comment (harrisonmeister) [reply docs/shared-content/infrastructure/add-new-environments.include.md]: Done in 08eddf8c5debdd7a60b38bab121d49f1ebb0dda1

**Pull request merged**